src/main/java/com/imaee/propinq/properties/controllers/implementations/PropertyTypeController.java

```diff
@@ -0,0 +1,66 @@
+package com.imaee.propinq.properties.controllers.implementations;
+
+import com.imaee.propinq.buildings.controllers.interfaces.IBuildingController;
+import com.imaee.propinq.properties.controllers.interfaces.IPropertyTypeController;
+import com.imaee.propinq.properties.controllers.requests.PropertyTypeRequest;
+import com.imaee.propinq.properties.controllers.responses.PropertyTypeResponse;
+import com.imaee.propinq.properties.services.interfaces.IPropertyTypeService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+import java.util.UUID;
+
+@RestController
+
+public class PropertyTypeController implements IPropertyTypeController {
+
+    private IPropertyTypeService propertyTypeService;
+
+    @Autowired
+    public PropertyTypeController(IPropertyTypeService propertyTypeService) {
+        this.propertyTypeService = propertyTypeService;
+    }
```

Comment:
En vez de usar el Autowired, podés usar directamente el `@AllArgsConstructor`. Esto te define un constructor para los argumentos que necesites. 

**Spring es capaz de inyectar los beans sin la necesidad del decorador aplicado sobre el constructor**. Magia

```suggestion
@AllArgsConstructor
public class PropertyTypeController implements IPropertyTypeController {

    private IPropertyTypeService propertyTypeService;

```